packages/xo-server/src/index.mjs

```diff
@@ -429,6 +430,22 @@ async function makeWebServerListen(
         opts.cert = pems.cert
         opts.key = pems.key
       }
+      if (opts.letsencrypt) {
+        opts.SNICallback = async (serverName, callback) => {
+          try {
+            const { cert, key } = await webServer.getCertificate(serverName, opts.cert, opts.key)
+            callback(
+              null,
+              tls.createSecureContext({
+                cert,
+                key,
+              })
```

Comment:
Maybe we can reuse the secure context if it's defined and still valid?

I'm afraid of the overhead if this method is called at each requests.